model_api/python/openvino/model_api/models/anomaly.py

```diff
@@ -0,0 +1,149 @@
+"""Definition for anomaly models.
+
+Note: This file will change when anomalib is upgraded in OTX. CVS-114640
+
+ Copyright (c) 2021-2023 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from __future__ import annotations
+
+from typing import Any
+
+import cv2
+import numpy as np
+
+from .image_model import ImageModel
+from .types import ListValue, NumericalValue, StringValue
+from .utils import AnomalyResult
+
+
+class AnomalyDetection(ImageModel):
+    __model__ = "AnomalyDetection"
+
+    def __init__(self, inference_adapter, configuration=None, preload=False):
+        super().__init__(inference_adapter, configuration, preload)
+        self._check_io_number(1, 1)
+        self.max: float
+        self.min: float
+        self.image_threshold: float
+        self.pixel_threshold: float
+        self.task: str
+        self.normalization_scale: float = self.max - self.min
+
+    def _setup_embedded_preprocessor(self, interpolation_mode: str = "LINEAR", dtype: type = ...) -> None:
+        """Override the parent method to set the dtype to float.
+
+        Args:
+            interpolation_mode (str, optional): Interpolation mode. Defaults to "LINEAR".
+            dtype (type, optional): Unused dtype. Overridden to float
+        """
+        super()._setup_embedded_preprocessor(interpolation_mode, dtype=float)
+
+    def preprocess(self, inputs: np.ndarray):
+        inputs = inputs / 255.0  # model expects inputs in range [0, 1]
+        return super().preprocess(inputs)
+
+    def postprocess(self, outputs: dict[str, np.ndarray], meta: dict[str, Any]):
+        """Post-processes the outputs and returns the results.
+
+        Args:
+            outputs (Dict[str, np.ndarray]): Raw model outputs
+            meta (Dict[str, Any]): Meta data containing the original image shape
+
+        Returns:
+            AnomalyResult: Results
+        """
+        anomaly_map: np.ndarray | None = None
```

Comment:
That's mostly up to you, it's just a suggestion. C++ version handles this by having empty cv::Mat. Analog to this in python is to have np.ndarray(0), not None.